modules/core/src/v2/coins/eos.ts

```diff
@@ -408,4 +438,268 @@ export class Eos extends BaseCoin {
+        }
+
+        if (!krsProviderConfig.supportedCoins.includes(this.getFamily())) {
+          throw new Error('specified key recovery service does not support recoveries for this coin');
+        }
+      }
+
+      const keys = [
+        this.validateKey({
+          key: userKey,
+          source: 'user',
+          passphrase: walletPassphrase,
+          isKrsRecovery,
+          isUnsignedSweep,
+        }),
+        this.validateKey({
+          key: backupKey,
+          source: 'backup',
+          passphrase: walletPassphrase,
+          isKrsRecovery,
+          isUnsignedSweep,
+        }),
+      ];
+      if (!this.isValidAddress(recoveryDestination)) {
+        throw new Error('Invalid destination address!');
+      }
+
+      return keys;
+    }).call(this);
+  }
+
+  getAccountFromNode(address): Bluebird<any> {
+    return co(function*() {
+      const nodeUrl = this.getPublicNodeUrl();
+      const result = yield this.bitgo.post(nodeUrl + '/v1/chain/get_account').send({ account_name: address });
+      if (!result || result.status !== 200) {
+        throw new Error('unable to fetch account from node: ' + nodeUrl);
+      }
+      return result.body;
+    }).call(this);
+  }
+
+  getTransactionHeadersFromNode(): Bluebird<any> {
+    return co(function*() {
+      const nodeUrl = this.getPublicNodeUrl();
+      const chainInfoResult = yield this.bitgo.post(nodeUrl + '/v1/chain/get_info').send();
+      if (!chainInfoResult || chainInfoResult.status !== 200) {
+        throw new Error('unable to fetch chain info from node: ' + nodeUrl);
+      }
+      const chainInfo = chainInfoResult.body;
+
+      const headBlockInfoResult = yield this.bitgo.post(nodeUrl + '/v1/chain/get_block').send({
+        block_num_or_id: chainInfo.head_block_num,
+      });
+      if (!headBlockInfoResult || headBlockInfoResult.status !== 200) {
+        throw new Error('unable to fetch chain info from node: ' + nodeUrl);
+      }
+      const expireSeconds = 3600; // maximum tx expire time of 1h
+      const chainDate = new Date(chainInfo.head_block_time + 'Z');
+      const expirationDate = new Date(chainDate.getTime() + expireSeconds * 1000);
+
+      return {
+        expiration: expirationDate.toISOString(),
+        ref_block_num: chainInfo.head_block_num & 0xffff,
+        ref_block_prefix: headBlockInfoResult.body.ref_block_prefix,
+      };
+    }).call(this);
+  }
+
+  getTransferAction = function({ recipient, sender, amount, memo }) {
+    return {
+      account: 'eosio.token',
+      name: 'transfer',
+      authorization: [
+        {
+          actor: sender,
+          permission: 'active',
+        },
+      ],
+      data: {
+        from: sender,
+        to: recipient,
+        quantity: `${this.baseUnitsToBigUnits(amount)} EOS`,
+        memo: !_.isNil(memo) ? memo : '', // Memo must be defined, set it to empty string if it is not
+      },
+    };
+  };
+
+  /**
+   * Sign a transaction with a key
+   * @param tx
```

Comment:
There are so many different types of `tx` objects in EOS I think we should be as specific as possible -- like "signableTxString" or something lol. "tx" could be the packed_trx, or this, or the stringified send JSON -- just too confusing